Makefile

```diff
@@ -734,7 +734,11 @@ OPTIMIZE    = -O0
 LTO_FLAGS   = $(OPTIMIZE)
 else
 OPTIMIZE    = -Os
+ifeq ($(TARGET),$(filter $(TARGET),$(F1_TARGETS)))
```

Comment:
Correct use of memory barriers (or volatile, but that brings other problems) is necessary. You only masked the problem, it may easily resurface later, and it may be extremely hard to find then.

You can debug  even with optimizations/lto enabled. Try `-O1`, that enables only optimization that don't interfere too much with debugging.
With little ARM assembly knowledge its possible to debug `-Os`, but moved code etc is a bit annoying ..